pubnub/lib/src/dx/channel/channel_history.dart

```diff
@@ -104,10 +106,14 @@ class ChannelHistory {
       _cursor = result.endTimetoken;
       _messages.addAll(await Future.wait(result.messages.map((message) async {
         if (_keyset.cipherKey != null) {
-          message['message'] = await _core.parser.decode(_core.crypto
-              .decrypt(_keyset.cipherKey!, message['message'] as String));
+          message['message'] = _keyset.cipherKey ==
```

Comment:
Could you explain to me what's the logic here?

Reply:
Possibility 1:  User has provided a custom cipherKey(or a keyset or one of the keyset from KeysetStore via using param) while calling history.
      In this case `_keyset(current one from param).cipherKey ==  _core.keysets.defaultKeyset.cipherKey` => `false`
          That means we need to use custom cipherKey - and so we perform `encryptWithKey`
          
Possibility 2:  User hasn't provided key which means _keyset.cipherKey is already defaulted to defaultKeyset's cipherkey 
      In this case `_keyset.cipherKey == _core.keysets.defaultKeyset.cipherKey`   => `true`
         That means we can use our `decrypt()` methods of cryptoModule because that cryptoModule is already initialised with defaultKeyset's cipherKey.